### PR TITLE
Fix Issue 18887 - inout badly described

### PR DESCRIPTION
**Pull request by RazvanN7**
Moved the inout section from functions to type qualifiers.

### CODE REVIEW

**Comment by dkorpel** [comment]
Did you change the content?

**Comment by RazvanN7** [comment]
Nope, simply copy-pasta. The only thing I added was a link in the inout function section to the new inout type qualifier section.

**Review by dkorpel — APPROVED** [review]

**Comment by jmh530** [comment]
@RazvanN7 I get that it is only copy-pasta, but it would be nice to have an example of the usage of `inout const`.

File: spec/function.dd [hunk 1]
```diff
@@ -698,139 +698,7 @@ $(H2 $(LNAME2 auto-ref-functions, Auto Ref Functions))
...
-            // rewritten to immutable(int)[]
-            immutable(int)[] z = bar(ia);
-        }
-        ---
-        )
-
-    $(P $(B Note:) Shared types cannot
-        be matched with `inout`.
-    )
+    $(P For extensive information read the $(LINK2 const3.hmtl#inout, $(D inout section of the type qualifiers documentation page)).)
```
Comment (ntrel) [inline spec/function.dd:701]: Typo hmtl. For LaTeX anchor I'd write this as:
```
$(DDSUBLINK2 spec/const3, inout, $(D inout) type qualifier).
```

**Pull request merged**